<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Haider Inam" />


<title>TCGA_luad_data_parser.Rmd</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">pair_con_select</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">TCGA_luad_data_parser.Rmd</h1>
<h4 class="author"><em>Haider Inam</em></h4>
<h4 class="date"><em>2/19/2019</em></h4>

</div>


<p><strong>Last updated:</strong> 2019-03-06</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>R Markdown file:</strong> up-to-date </summary></p>
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(20190211)</code> </summary></p>
<p>The command <code>set.seed(20190211)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/haiderinam/pair_con_select/tree/b7d812d4ab27cd7a70ba750f39a0edd13091bad5" target="_blank">b7d812d</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  code/alldata_compiler.R
    Untracked:  code/contab_maker.R
    Untracked:  code/mut_excl_genes_datapoints.R
    Untracked:  code/mut_excl_genes_generator.R
    Untracked:  code/quadratic_solver.R
    Untracked:  code/simresults_generator.R
    Untracked:  data/All_Data_V2.csv
    Untracked:  data/alkati_growthcurvedata.csv
    Untracked:  data/alkati_growthcurvedata_popdoublings.csv
    Untracked:  data/alkati_melanoma_vemurafenib_figure_data.csv
    Untracked:  data/all_data.csv
    Untracked:  data/tcga_luad_expression/
    Untracked:  data/tcga_skcm_expression/
    Untracked:  docs/figure/Filteranalysis.Rmd/
    Untracked:  docs/figure/baf3_alkati_transformations.Rmd/
    Untracked:  output/alkati_filtercutoff_allfilters.csv
    Untracked:  output/alkati_luad_exonimbalance.pdf
    Untracked:  output/alkati_mtn_pval_fig2B.pdf
    Untracked:  output/alkati_skcm_exonimbalance.pdf
    Untracked:  output/all_data_luad.csv
    Untracked:  output/all_data_luad_egfr.csv
    Untracked:  output/all_data_skcm.csv
    Untracked:  output/baf3_alkati_figure_deltaadjusted_doublings.pdf
    Untracked:  output/baf3_barplot.pdf
    Untracked:  output/baf3_elisa_barplot.pdf
    Untracked:  output/egfr_luad_exonimbalance.pdf
    Untracked:  output/fig2b2_filtercutoff_atinras_totalalk.pdf
    Untracked:  output/fig2b_filtercutoff_atibraf.pdf
    Untracked:  output/fig2b_filtercutoff_atinras.pdf
    Untracked:  output/luad_alk_exon_expression.csv
    Untracked:  output/luad_egfr_exon_expression.csv
    Untracked:  output/melanoma_vemurafenib_fig.pdf
    Untracked:  output/skcm_alk_exon_expression.csv

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<details> <summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/haiderinam/pair_con_select/aff917b4079ef57366ed90adf6c4b8d4a31383af/docs/TCGA_luad_data_parser.html" target="_blank">aff917b</a>
</td>
<td style="text-align:left;">
haiderinam
</td>
<td style="text-align:left;">
2019-02-20
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/haiderinam/pair_con_select/blob/6fec6220ee12e0b3a246b129a30ce3c874dc3b51/analysis/TCGA_luad_data_parser.Rmd" target="_blank">6fec622</a>
</td>
<td style="text-align:left;">
haiderinam
</td>
<td style="text-align:left;">
2019-02-20
</td>
<td style="text-align:left;">
Added EGFR Expression Analysis
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/haiderinam/pair_con_select/0b5f5cb94c139b5c731b45b4f5ae45dd07138e02/docs/TCGA_luad_data_parser.html" target="_blank">0b5f5cb</a>
</td>
<td style="text-align:left;">
haiderinam
</td>
<td style="text-align:left;">
2019-02-19
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/haiderinam/pair_con_select/blob/f3edec6a1af73b881e4b47ec148843460b08ae2e/analysis/TCGA_luad_data_parser.Rmd" target="_blank">f3edec6</a>
</td>
<td style="text-align:left;">
haiderinam
</td>
<td style="text-align:left;">
2019-02-19
</td>
<td style="text-align:left;">
Published Analysis on ALK expression levels in LUAD
</td>
</tr>
</tbody>
</table>
</ul>
<p></details></p>
<hr />
<div id="this-code-essentially-processes-luad-tcga-data" class="section level2">
<h2>This code essentially processes LUAD TCGA data:</h2>
<ol style="list-style-type: decimal">
<li>Mutation Data</li>
<li>RSEM Normalized</li>
<li>mRNA Count DAta</li>
<li>Exon Expression data<br />
Once it processes these data, it makes a merged all_data_luad.csv file for ALK that contains:</li>
<li>RSEM</li>
<li>RPKM (exon 20-29/exon 1-19 expression)</li>
<li>Count data</li>
<li>EGFR and KRAS Mutations<br />
Please note: there are two chunks with eval=F. Therefore, turn EVAL=T when running these for the first time</li>
</ol>
</div>
<div id="luad-mutation-data-grabbing-egfr-and-kras-for-230-patients." class="section level2">
<h2><strong>LUAD Mutation Data: Grabbing EGFR and KRAS for 230 patients.</strong></h2>
<pre class="r"><code>x1=list.dirs(&quot;data/tcga_luad_expression/luad_mutation_data&quot;,full.names=TRUE)#Lists all files in the current working directory x

Data_list=list.files(x1[1],pattern=&quot;^TCGA-[A-Za-z0-9]{2}-[A-Za-z0-9]{4}-[A-Za-z0-9]{2}.hg19.oncotator.hugo_entrez_remapped.maf.txt*&quot;, ignore.case=F)#lists the files in the folder

  patmat=matrix(nrow=length(Data_list),ncol=3)# This initializes the storage matrix
  for (i in 1:length(Data_list)){
    patdat=read.table(paste(x1,&quot;/&quot;,Data_list[i],sep=&quot;&quot;),stringsAsFactors=FALSE,header=TRUE, sep=&quot;\t&quot;,fill=TRUE,quote = &quot;&quot;)#opens each file as the loop progresses
    egfr=patdat%&gt;%filter(Hugo_Symbol==&quot;EGFR&quot;,Variant_Classification!=&quot;Silent&quot;)
    kras=patdat%&gt;%filter(Hugo_Symbol==&quot;KRAS&quot;,Variant_Classification!=&quot;Silent&quot;)

    #This essentially says that if you can&#39;t find the mutant, enter NaN. If you find two mutants, then search for the major transforming mutation (e.g. BrafV600E). Non of the &gt;1 mutations are the transforming mutation, just select the first one
      if(nrow(egfr)&gt;=2){
      if(egfr$Protein_Change%in%&quot;p.T790M&quot;){
        egfr=egfr%&gt;%filter(Protein_Change==&quot;p.T790M&quot;)
      } else{
          egfr=egfr[1,]
        }
      } else if(nrow(egfr)==0){
        egfr[1,]=&quot;p.NaN&quot;
      }
    if(nrow(kras)&gt;=2){
      if(kras$Protein_Change%in%&quot;p.G12D&quot;){
        kras=kras%&gt;%filter(Protein_Change==&quot;p.G12D&quot;)
      } else{
          kras=kras[1,]
        }
      } else if(nrow(kras)==0){
        kras[1,]=&quot;p.NaN&quot;
      }

    # missense=nrow(patdat[patdat$Variant_Classification==&quot;Missense_Mutation&quot;,])#counts missense mutations by identifying the number of rows in a   
    patmat[i,1]=Data_list[i]#Record the Patient ID from the file name
    patmat[i,2]=egfr$Protein_Change
    patmat[i,3]=kras$Protein_Change
    }</code></pre>
<pre><code>Warning in if (egfr$Protein_Change %in% &quot;p.T790M&quot;) {: the condition has
length &gt; 1 and only the first element will be used

Warning in if (egfr$Protein_Change %in% &quot;p.T790M&quot;) {: the condition has
length &gt; 1 and only the first element will be used

Warning in if (egfr$Protein_Change %in% &quot;p.T790M&quot;) {: the condition has
length &gt; 1 and only the first element will be used

Warning in if (egfr$Protein_Change %in% &quot;p.T790M&quot;) {: the condition has
length &gt; 1 and only the first element will be used

Warning in if (egfr$Protein_Change %in% &quot;p.T790M&quot;) {: the condition has
length &gt; 1 and only the first element will be used

Warning in if (egfr$Protein_Change %in% &quot;p.T790M&quot;) {: the condition has
length &gt; 1 and only the first element will be used

Warning in if (egfr$Protein_Change %in% &quot;p.T790M&quot;) {: the condition has
length &gt; 1 and only the first element will be used</code></pre>
<pre><code>Warning in if (kras$Protein_Change %in% &quot;p.G12D&quot;) {: the condition has
length &gt; 1 and only the first element will be used</code></pre>
<pre><code>Warning in if (egfr$Protein_Change %in% &quot;p.T790M&quot;) {: the condition has
length &gt; 1 and only the first element will be used

Warning in if (egfr$Protein_Change %in% &quot;p.T790M&quot;) {: the condition has
length &gt; 1 and only the first element will be used

Warning in if (egfr$Protein_Change %in% &quot;p.T790M&quot;) {: the condition has
length &gt; 1 and only the first element will be used</code></pre>
<pre class="r"><code>patframe=data.frame(patmat)#Turn storage matrix into data frame
colnames(patframe)[1:3]=c(&quot;Patid&quot;,&quot;EGFR&quot;,&quot;KRAS&quot;)#Rename the columns
# write.csv(patframe,&quot;patients_tally_muttype2.csv&quot;)# Record data frame as a CSV and write to the working directory

#Grabbing Patient Names so that they can be used to merge with exon data later
alk_mutated_data=patframe
alk_mutated_data$Patid=substring(alk_mutated_data$Patid,first = 1,last = 12)
###Removing &quot;p.&quot; from names of mutants:
alk_mutated_data$EGFR=unlist(sub(&quot;p.&quot;,&quot;&quot;,alk_mutated_data$EGFR))
alk_mutated_data$KRAS=unlist(sub(&quot;p.&quot;,&quot;&quot;,alk_mutated_data$KRAS))
head(alk_mutated_data)</code></pre>
<pre><code>         Patid  EGFR KRAS
1 TCGA-05-4249   NaN G12C
2 TCGA-05-4382 R222L  NaN
3 TCGA-05-4384   NaN  NaN
4 TCGA-05-4389   NaN  NaN
5 TCGA-05-4390   NaN G12V
6 TCGA-05-4395   NaN G12V</code></pre>
<p>LUAD Genes RSEM</p>
<pre class="r"><code>rsemdatanormalized=read.table(&quot;data/tcga_luad_expression/luadrsemdata/gdac.broadinstitute.org_LUAD.Merge_rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__RSEM_genes_normalized__data.Level_3.2016012800.0.0/LUAD.rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__RSEM_genes_normalized__data.data.txt&quot;,sep = &quot;\t&quot;,header = T,stringsAsFactors = F)
alk_rsem=data.frame(t(rsemdatanormalized[grepl(&quot;^alk\\|&quot;,rsemdatanormalized$Hybridization.REF,ignore.case = T),])[-1,])
#410 of the 577 patients have an RSEM higher than 410
colnames(alk_rsem)[1]=&quot;RSEM_normalized&quot;
alk_rsem$Patid=rownames(alk_rsem)
#Standardizing Patid Names
alk_rsem$Patid=substring(alk_rsem$Patid,first = 1,last = 12)
alk_rsem$Patid=gsub(&quot;\\.&quot;,&quot;-&quot;,alk_rsem$Patid)

# # As Character
alk_rsem[colnames(alk_rsem)] &lt;- lapply(alk_rsem[colnames(alk_rsem)],as.character)
# # As Numeric: Converting from list to numeric
alk_rsem$RSEM_normalized=unlist(alk_rsem$RSEM_normalized)
alk_rsem$RSEM_normalized=as.numeric(alk_rsem$RSEM_normalized)</code></pre>
<p>LUAD Count data:</p>
<pre class="r"><code>#Non-normalized:
gene_expression_data=read.table(&quot;data/tcga_luad_expression/luadgeneexpression/gdac.broadinstitute.org_LUAD.Merge_rnaseq__illuminahiseq_rnaseq__unc_edu__Level_3__gene_expression__data.Level_3.2016012800.0.0/LUAD.rnaseq__illuminahiseq_rnaseq__unc_edu__Level_3__gene_expression__data.data.txt&quot;,sep = &quot;\t&quot;,header = T,stringsAsFactors = F)
#Normalized
# gene_expression_data=read.table(,sep = &quot;\t&quot;,header = T,stringsAsFactors = F)

  #Finding Alk
  alk_gene_exp=rbind(gene_expression_data[1,],gene_expression_data[grepl(&quot;^alk\\|&quot;,gene_expression_data$Hybridization.REF,ignore.case = T),])
  #Removing Columns for Median_length_normalized and RPKM
  t_alk_gene_exp=data.frame(t(alk_gene_exp[,grepl(&quot;raw_count&quot;,alk_gene_exp[1,])]))
  
  #Counting patients with raw reads &gt;500
  # sum(as.numeric(as.numeric(as.character(t_alk_gene_exp$X580))&gt;=500))
#ONLY 6 PATIENTS HAVE RAW COUNTS OF &gt;500</code></pre>
<p>LUAD Exon RPKM This creates a .csv file and only needs to be run once.</p>
<pre class="r"><code># rm(list=ls())#Clears workspace
exondatacomb=read.table(&quot;data/tcga_luad_expression/luadexondatacomb/gdac.broadinstitute.org_LUAD.Merge_rnaseq__illuminahiseq_rnaseq__unc_edu__Level_3__exon_expression__data.Level_3.2016012800.0.0/LUAD.rnaseq__illuminahiseq_rnaseq__unc_edu__Level_3__exon_expression__data.data.txt&quot;,stringsAsFactors=FALSE,header=TRUE, sep=&quot;\t&quot;,fill=TRUE)
# head(exondatacomb)

#Chromosome 2
exondatachr2=exondatacomb[grep(&quot;^chr2:&quot;,exondatacomb$Hybridization.REF),] #i.e. it starts with chromosome 2
#Alk within Chromosome 2
# The exon locations were found on ensembl here https://useast.ensembl.org/Homo_sapiens/Transcript/Exons?db=core;g=ENSG00000171094;r=2:29192774-29921566;t=ENST00000389048
##These start at chr2:29415641-29416788:-
exondatachr2alk=exondatacomb[c(26031:26059),]

# # write.table(exondatachr2alk,&#39;exondatachr2alk.csv&#39;)
# exondatachr2alk=read.csv(&quot;exondatachr2alk.csv&quot;,stringsAsFactors = F,header = T,sep = &quot;&quot;,fill = T)
#Adding Names for Exons
exondatachr2alk$exon=c(29:1)
alldataalk=exondatachr2alk[,c(488,c(2:487))]
#Switching up order
alldataalk2=alldataalk[c(29:1),]

#Making the dataframe of a numeric type so that analysis can be carried out on it.
# As Character
alldataalk2[colnames(alldataalk2)] &lt;- lapply(alldataalk2[colnames(alldataalk2)],as.character)
# As Numeric
alldataalk2[colnames(alldataalk2)] &lt;- lapply(alldataalk2[colnames(alldataalk2)],as.numeric)

#Getting the correct column names for alldataalk2
# alldataalk2[1,]
colnames_exondata=exondatacomb[1,]
colnames(colnames_exondata)=colnames(alldataalk2)

alldataalk2=rbind(colnames_exondata,alldataalk2) #Adding first row that contains names of measurements such as RPKM, RSEM, Counts
write.table(alldataalk2,&#39;output/luad_alk_exon_expression.csv&#39;)

#I used this code to find the length of exons and compare these to the lengths of the exons on Ensembl. I had to calculate exon lengths because annotations in this file and annotations in enseml weren&#39;t the same.
# trunc_names=gsub(&quot;chr2:|:\\+|:\\-&quot;,&quot;&quot;,exondatachr2$Hybridization.REF)
# ##Code to get the length of each exon:
# names=exondatachr2$Hybridization.REF
# trunc_names2=gsub(&quot;\\-&quot;,&quot;&quot;,trunc_names)
# trunc_names2=gsub(&quot;chr2:&quot;,&quot;&quot;,trunc_names)
# start=sapply(strsplit(trunc_names,&quot;-&quot;),&quot;[&quot;,1)
# end=sapply(strsplit(trunc_names,&quot;-&quot;),&quot;[&quot;,2)
# positions=data.frame(start,end,names)
# positions[,c(1,2)]=lapply(positions[,c(1,2)],as.character)
# positions[,c(1,2)]=lapply(positions[,c(1,2)],as.numeric)
# positions$net=positions$end-positions$start</code></pre>
<div id="obtaining-rpkm-count-data-from-luad-exon-data.-followed-by-merging-with-rsem-data" class="section level3">
<h3>Obtaining RPKM, Count data from LUAD Exon Data. Followed by merging with RSEM data</h3>
<pre class="r"><code>alldataalk2=read.csv(&quot;output/luad_alk_exon_expression.csv&quot;,stringsAsFactors = F,header = T,sep = &quot;&quot;,fill = T)
#Getting Count Data
alldataalk2_count=cbind(alldataalk2$exon,alldataalk2[,grepl(&quot;raw_counts&quot;,alldataalk2[1,])])[-1,]
# As Character
alldataalk2_count[colnames(alldataalk2_count)] &lt;-lapply(alldataalk2_count[colnames(alldataalk2_count)],as.character)
# As Numeric
alldataalk2_count[colnames(alldataalk2_count)] &lt;- lapply(alldataalk2_count[colnames(alldataalk2_count)],as.numeric)
#Sum exons 1:29
alk_count_data=data.frame(t(data.frame(lapply(alldataalk2_count[c(1:29),],sum))[,-1])) #Not sure if lapply is the right thing to use here. Really messed up way of summing indices in dataframe
colnames(alk_count_data)=&quot;mRNA_count&quot;

alldataalk2_medianlength=cbind(alldataalk2$exon,alldataalk2[,grepl(&quot;median_length&quot;,alldataalk2[1,])])
# As Character
alldataalk2_medianlength=alldataalk2_medianlength[-1,] #Removing the first row. May be unnecessary in the future
alldataalk2_medianlength[colnames(alldataalk2_medianlength)] &lt;- lapply(alldataalk2_medianlength[colnames(alldataalk2_medianlength)],as.character)
# As Numeric
alldataalk2_medianlength[colnames(alldataalk2_medianlength)] &lt;- lapply(alldataalk2_medianlength[colnames(alldataalk2_medianlength)],as.numeric)
#Sum exons 1:29
alk_medianlength_data=data.frame(t(data.frame(lapply(alldataalk2_medianlength[c(1:29),],sum))[,-1])) #Removing sum of exons lol
colnames(alk_medianlength_data)=&quot;medianlength&quot;

#Getting RPKM
alldataalk2_RPKM=cbind(alldataalk2$exon,alldataalk2[,grepl(&quot;RPKM&quot;,alldataalk2[1,])])
# As Character
alldataalk2_RPKM=alldataalk2_RPKM[-1,] #Removing the first row. May be unnecessary in the future
alldataalk2_RPKM[colnames(alldataalk2_RPKM)] &lt;- lapply(alldataalk2_RPKM[colnames(alldataalk2_RPKM)],as.character)
# As Numeric
alldataalk2_RPKM[colnames(alldataalk2_RPKM)] &lt;- lapply(alldataalk2_RPKM[colnames(alldataalk2_RPKM)],as.numeric)
alk_RPKM_data=data.frame(cbind(lapply(alldataalk2_RPKM[c(1:19),],mean),lapply(alldataalk2_RPKM[c(20:29),],mean))[-1,])
colnames(alk_RPKM_data)=c(&quot;mean_RPKM_1.19&quot;,&quot;mean_RPKM_20.29&quot;)
# As Character
alk_RPKM_data[colnames(alk_RPKM_data)] &lt;- lapply(alk_RPKM_data[colnames(alk_RPKM_data)],as.character)
# As Numeric
alk_RPKM_data[colnames(alk_RPKM_data)] &lt;- lapply(alk_RPKM_data[colnames(alk_RPKM_data)],as.numeric)
#  Calculating Ratios of exon RPKM means
alk_RPKM_data$Ratio20.29=alk_RPKM_data$mean_RPKM_20.29/alk_RPKM_data$mean_RPKM_1.19

#Changing rownames (patient_ids) to become the same between each other
rownames(alk_RPKM_data)=substring(rownames(alk_RPKM_data),first=1,last=28)
rownames(alk_medianlength_data)=substring(rownames(alk_medianlength_data),first=1,last=28)
alk_RPKM_data$Patid=rownames(alk_RPKM_data)
alk_medianlength_data$Patid=rownames(alk_medianlength_data)
alk_count_data$Patid=rownames(alk_count_data)

mergetemp=merge(alk_RPKM_data,alk_count_data,by=&quot;Patid&quot;)
alk_exon_data=merge(mergetemp,alk_medianlength_data,by=&quot;Patid&quot;)
#Transforming the Patids so that they&#39;re compatible with the Patids in the mutation data
alk_exon_data$Patid=substring(alk_exon_data$Patid,first = 1,last = 12)
#Since the names for exon data are not the same format as the mutation data, we&#39;re gonna change that here
alk_exon_data$Patid=gsub(&quot;\\.&quot;,&quot;-&quot;,alk_exon_data$Patid)
alkati_merged_data=merge(alk_exon_data,alk_mutated_data,by=&quot;Patid&quot;,all=T)
alkati_merged_data=merge(alkati_merged_data,alk_rsem,by=&quot;Patid&quot;,all = T)
# alkati_merged_data=merge(alk_exon_data,alk_mutated_data,by=&quot;Patid&quot;)
# alkati_merged_data=merge(alkati_merged_data,alk_rsem,by=&quot;Patid&quot;)

###Now adding ALK hits to the data based on filters by Wiesner et al
###2/15 note: use the TCGA data sorter to just process your data
# alk_data=read.csv(&quot;../data/all_data.csv&quot;,stringsAsFactors = F)
# alldata=tcgadatasorter(&quot;data/all_data.csv&quot;,meanRPKM,100,500)
alkati_merged_data_alkati=alkati_merged_data%&gt;%
  group_by(Patid,mean_RPKM_1.19,mean_RPKM_20.29,Ratio20.29, mRNA_count,EGFR,KRAS,RSEM_normalized)%&gt;%
  summarize(ATI=as.numeric(mRNA_count&gt;=100&amp;Ratio20.29&gt;10&amp;RSEM_normalized&gt;=80)[1])

write.csv(alkati_merged_data_alkati,&quot;output/all_data_luad.csv&quot;)</code></pre>
<p>Making ALK Expression the plots:</p>
<pre class="r"><code>alkati_merged_data=read.csv(&quot;output/all_data_luad.csv&quot;)
alkati_merged_data$alkati=0
alkati_merged_data$alkati[alkati_merged_data$Ratio&gt;=10&amp;alkati_merged_data$mRNA_count&gt;=500&amp;alkati_merged_data$RSEM_normalized&gt;=80]=1
alkati_merged_data$alkati=factor(alkati_merged_data$alkati,levels=c(&quot;1&quot;,&quot;0&quot;))

  ggplot(alkati_merged_data,aes(x=mean_RPKM_1.19, y=mean_RPKM_20.29,color=factor(alkati)))+
    geom_abline(size=1)+
    geom_point(size=4)+
    ####Had to add this line to not overplot the alkati datapoint- Haider 1/31/19
    geom_point(data=alkati_merged_data[alkati_merged_data$alkati==1,],aes(x=mean_RPKM_1.19, y=mean_RPKM_20.29,color=factor(alkati)),size=4)+
    scale_x_continuous(trans = &quot;log10&quot;,name=&quot;Exon 1:19 RPKM&quot;,breaks=c(1e-2,1e0,1e2),labels = parse(text = c(&quot;10^-2&quot;,&quot;10^0&quot;,&quot;10^2&quot;)),limits = c(1e-3,1e3))+
    scale_y_continuous(trans = &quot;log10&quot;,name=&quot;Exon 20:29 RPKM&quot;,breaks=c(1e-2,1e0,1e2),labels = parse(text = c(&quot;10^-2&quot;,&quot;10^0&quot;,&quot;10^2&quot;)),limits = c(1e-3,1e3))+
    scale_color_brewer(palette=&quot;Set1&quot;,name=&quot;ALKATI&quot;,labels=c(&quot;Yes&quot;, &quot;No&quot;))+
    cleanup+
    theme(plot.title = element_text(hjust=.5),
          text = element_text(size=24,face = &quot;bold&quot;),
          axis.title = element_text(face=&quot;bold&quot;,size=&quot;24&quot;),
          axis.text=element_text(face=&quot;bold&quot;,size=&quot;24&quot;,colour = &quot;black&quot;))+
    theme(legend.key.size = unit(30,&quot;pt&quot;))</code></pre>
<pre><code>Warning: Transformation introduced infinite values in continuous x-axis</code></pre>
<pre><code>Warning: Transformation introduced infinite values in continuous y-axis</code></pre>
<pre><code>Warning: Removed 391 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/TCGA_luad_data_parser.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-7-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/haiderinam/pair_con_select/blob/0b5f5cb94c139b5c731b45b4f5ae45dd07138e02/docs/figure/TCGA_luad_data_parser.Rmd/unnamed-chunk-7-1.png" target="_blank">0b5f5cb</a>
</td>
<td style="text-align:left;">
haiderinam
</td>
<td style="text-align:left;">
2019-02-19
</td>
</tr>
</tbody>
</table>
<p></details></p>
<pre class="r"><code>ggsave(&quot;output/alkati_luad_exonimbalance.pdf&quot;,width =12 ,height =10 ,units = &quot;in&quot;,useDingbats=F)</code></pre>
<pre><code>Warning: Transformation introduced infinite values in continuous x-axis</code></pre>
<pre><code>Warning: Transformation introduced infinite values in continuous y-axis</code></pre>
<pre><code>Warning: Removed 391 rows containing missing values (geom_point).</code></pre>
<pre class="r"><code>#Testing if both kinase and ALK expression are different
ks.test(alkati_merged_data$mean_RPKM_1.19,alkati_merged_data$mean_RPKM_20.29)</code></pre>
<pre><code>Warning in ks.test(alkati_merged_data$mean_RPKM_1.19,
alkati_merged_data$mean_RPKM_20.29): p-value will be approximate in the
presence of ties</code></pre>
<pre><code>
    Two-sample Kolmogorov-Smirnov test

data:  alkati_merged_data$mean_RPKM_1.19 and alkati_merged_data$mean_RPKM_20.29
D = 0.65401, p-value &lt; 2.2e-16
alternative hypothesis: two-sided</code></pre>
<pre class="r"><code>###We observed a significant difference between the distribution for the 20-29 exons and the 1-19 exons The reported p-value was 2-16.</code></pre>
</div>
<div id="statistical-test-to-see-if-alk-kinase-domain-expression-was-significantly-higher-than-other-domains" class="section level3">
<h3>Statistical test to see if ALK kinase domain expression was significantly higher than other domains</h3>
<pre class="r"><code>#Testing if both kinase and ALK expression are different
ks.test(alkati_merged_data$mean_RPKM_1.19,alkati_merged_data$mean_RPKM_20.29)</code></pre>
<pre><code>Warning in ks.test(alkati_merged_data$mean_RPKM_1.19,
alkati_merged_data$mean_RPKM_20.29): p-value will be approximate in the
presence of ties</code></pre>
<pre><code>
    Two-sample Kolmogorov-Smirnov test

data:  alkati_merged_data$mean_RPKM_1.19 and alkati_merged_data$mean_RPKM_20.29
D = 0.65401, p-value &lt; 2.2e-16
alternative hypothesis: two-sided</code></pre>
<pre class="r"><code>###We observed a significant difference between the distribution for the 20-29 exons and the 1-19 exons The reported p-value was 2-16.
###The p-value from a Chi-sq test was 2.2e-16 too
    #for all ALK data, not ALKATI
ov_expr_obs=c(sum(as.numeric(alkati_merged_data$Ratio20.29&gt;1),na.rm = T),
              dim(alkati_merged_data)[1]-sum(as.numeric(alkati_merged_data$Ratio20.29&gt;1),na.rm = T))
ov_expr_expected=c(round(dim(alkati_merged_data)[1]/2),
                   round(dim(alkati_merged_data)[1]/2))
overexpression=data.frame(rbind(ov_expr_obs,ov_expr_expected))
colnames(overexpression)=c(&quot;Yes&quot;,&quot;No&quot;)
chisq.test(overexpression)</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  overexpression
X-squared = 27.652, df = 1, p-value = 1.452e-07</code></pre>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.2 (2018-12-20)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Mojave 10.14.3

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  grid      stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] bindrcpp_0.2.2      ggsignif_0.4.0      usethis_1.4.0      
 [4] devtools_2.0.1      RColorBrewer_1.1-2  reshape2_1.4.3     
 [7] ggplot2_3.1.0       doParallel_1.0.14   iterators_1.0.10   
[10] foreach_1.4.4       dplyr_0.7.8         VennDiagram_1.6.20 
[13] futile.logger_1.4.3 workflowr_1.1.1     tictoc_1.0         
[16] knitr_1.21         

loaded via a namespace (and not attached):
 [1] tidyselect_0.2.5     xfun_0.4             remotes_2.0.2       
 [4] purrr_0.3.0          colorspace_1.4-0     htmltools_0.3.6     
 [7] yaml_2.2.0           rlang_0.3.1          pkgbuild_1.0.2      
[10] R.oo_1.22.0          pillar_1.3.1         glue_1.3.0          
[13] withr_2.1.2          R.utils_2.7.0        sessioninfo_1.1.1   
[16] lambda.r_1.2.3       bindr_0.1.1          plyr_1.8.4          
[19] stringr_1.3.1        munsell_0.5.0        gtable_0.2.0        
[22] R.methodsS3_1.7.1    codetools_0.2-16     evaluate_0.12       
[25] memoise_1.1.0        callr_3.1.1          ps_1.3.0            
[28] Rcpp_1.0.0           backports_1.1.3      scales_1.0.0        
[31] formatR_1.5          desc_1.2.0           pkgload_1.0.2       
[34] fs_1.2.6             digest_0.6.18        stringi_1.2.4       
[37] processx_3.2.1       rprojroot_1.3-2      cli_1.0.1           
[40] tools_3.5.2          magrittr_1.5         lazyeval_0.2.1      
[43] tibble_2.0.1         futile.options_1.0.1 crayon_1.3.4        
[46] whisker_0.3-2        pkgconfig_2.0.2      prettyunits_1.0.2   
[49] assertthat_0.2.0     rmarkdown_1.11       R6_2.3.0            
[52] git2r_0.24.0         compiler_3.5.2      </code></pre>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
